modules/types_proto/proto/hephaestus/types/proto/bounds.proto

```diff
@@ -0,0 +1,22 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+syntax = "proto3";
+
+package heph.types.proto;
+
+import "hephaestus/types/proto/numeric_value.proto";
+
+enum BoundsType {
+  INCLUSIVE = 0;
+  LEFT_OPEN = 1;   
+  RIGHT_OPEN = 2;  
+  OPEN = 3;         
+}
+
+message Bounds {
+  NumericValue lower_bound = 1;
+  NumericValue upper_bound = 2;
+  BoundsType bounds_type = 3;
+}
```

Comment:
```suggestion
message Bounds {
enum Type {
  INCLUSIVE = 0;
  LEFT_OPEN = 1;   
  RIGHT_OPEN = 2;  
  OPEN = 3;         
}
  NumericValue lower_bound = 1;
  NumericValue upper_bound = 2;
  BoundsType bounds_type = 3;
}
```
wdyt?

Reply:
if it is posisble to define enums inside messages, yes I agree!

Reply:
So this code works. However, there are two downsides.
1. I do not want to do the same in our source code, as Bounds is templated, but BoundsType is not, which simplifies things. I would not want to change that.
2. Readability. By removing the explicit enum, protobuf decays all enum values into constants of Bounds, which reduces readability, i.e.
```
case BoundsType::INCLUSIVE:
  return proto::Bounds::INCLUSIVE;
```

Wdyt? Keep the current version or change to the new version? In both cases, I would keep `BoundsType` outside of `Bounds<T>`.

Reply:
For readability and linting purposes, we keep enum BoundsType separate.